repos/system_upgrade/common/actors/openssl/migrateopensslconf/actor.py

```diff
@@ -0,0 +1,26 @@
+from leapp.actors import Actor
+from leapp.libraries.actor import migrateopensslconf
+from leapp.tags import ApplicationsPhaseTag, IPUWorkflowTag
+
+
+class MigrateOpenSslConf(Actor):
+    """
+    Enforce the target default configuration file to be used.
+
+    If the /etc/pki/tls/openssl.cnf has been modified and openssl.cnf.rpmnew
+    file is created, backup the original one and replace it by the new default.
+
+    tl;dr:
+    if the file is modified; then
```

Comment:
This is simple but incorrect way how to approach it. `rpmsave` and `rpmnew` files are created during `rpm` transaction, which means that if a config file has been updated AFTER the latest update, these files won't exist.
I am afraid there is no simple way how to restore the original config file (I mean other than extracting it from RPM file) but better to consult RPM folks.

Reply:
@kkaarreell you are right, and the code treats that - the tl;dr; is not exact. If I understand correctly your concern, it should be ok as:
* if the file has been modified before the transaction, it rpmnew file will exist
* if the file has been modified after the transaction (so the rpmnew is not created during the transaction), we consider the change as intentional and we do nothing

In this part, we already cannot rely on having the rpm locally (and we cannot rely on internet), so taking it from the package is not a point here. We could think about having the default file stored as a part of the actor, however that will not cover cases when the file is changed (z-stream, differences  between minor versions when nobody track this, etc.). We need to take into account, that nothing prevents customer to change the file anyway after this piece of code is executed.

Reply:
Discussed over video call. Current approach is ok from the point of possible missing *.rpmnew file.